File: app/src/main/java/by/overpass/gather/ui/base/BackPressActivity.java
package by.overpass.gather.ui.base;

import androidx.lifecycle.ViewModel;

import com.annimon.stream.Stream;

public abstract class BackPressActivity<VM extends ViewModel> extends BaseActivityKt<VM> {

    @Override
    public void onBackPressed() {
        if (!Stream.of(getSupportFragmentManager().getFragments())
                .filter(fragment -> fragment instanceof BackPressFragment)
                .map(fragment -> (BackPressFragment) fragment)
                .anyMatch(BackPressFragment::handleBackPress)) {
            super.onBackPressed();
        }
    }
}

File: app/src/main/java/by/overpass/gather/ui/base/BackPressActivity.kt
package by.overpass.gather.ui.base

import androidx.lifecycle.ViewModel

abstract class BackPressActivity<VM : ViewModel> : BaseActivityKt<VM>() {

    override fun onBackPressed() {
        if (!supportFragmentManager.fragments
                        .filterIsInstance<BackPressFragment>()
                        .any { it.handleBackPress() }) {
            super.onBackPressed()
        }
    }
}
